net/libfab/libfab.c

```diff
@@ -0,0 +1,3455 @@
+			}
+		}
+
+		libfab_bufq_process(tm);
+		if (m0_time_sub(m0_time_now(), tm->ftm_tmout_check) >=
+								 m0_time(1,0))
+			libfab_tm_buf_timeout(tm);
+		libfab_tm_buf_done(tm);
+
+		M0_ASSERT(libfab_tm_invariant(tm));
+		libfab_tm_unlock(tm);
+	}
+}
+
+/** 
+ * Converts network end-point to its libfabric structure.
+ */
+static inline struct m0_fab__ep *libfab_ep(struct m0_net_end_point *net)
+{
+	return container_of(net, struct m0_fab__ep, fep_nep);
+}
+
+/**
+ * Compares the endpoint name with the passed name string and
+ * returns true if equal, or else returns false
+ * If name is null then the ipaddress and port fields are compared to check for
+ * a matching endpoint.
+ */
+static bool libfab_ep_cmp(struct m0_fab__ep *ep, const char *name,
+			  uint64_t *ep_name_n)
+{
+	bool ret = false;
+	M0_PRE(ep != NULL);
+	if (name == NULL) {
+		if (*ep_name_n == ep->fep_name_n)
+			ret = true;
+	} else {
+		if (strcmp(ep->fep_name_p.fen_str_addr, name) == 0)
+			ret = true;
+	}
+	return ret;
+}
+
+/**
+ * Search for the ep in the existing ep list using one of the following -
+ *   1) Name in str format      OR
+ *   2) ipaddr and port 
+ * If found then return the ep structure, or else create a new endpoint
+ * with the name
+ * Returns 0 if the endpoint is found/created succesfully,
+ * else returns error code.
+ */
+static int libfab_ep_find(struct m0_net_transfer_mc *tm, const char *name,
+			  struct m0_fab__ep_name *epn,
+			  struct m0_net_end_point **epp)
+{
+	struct m0_net_end_point  *net;
+	struct m0_fab__ep        *ep;
+	struct m0_fab__active_ep *aep;
+	struct m0_fab__tm        *ma;
+	uint64_t                  ep_name_n = 0;
+	char                      ep_str[LIBFAB_ADDR_STRLEN_MAX + 9];
+	char                     *wc = NULL;
+	int                       rc = 0;
+
+	if (epn != NULL)
+		libfab_ep_pton(epn, &ep_name_n);
+
+	M0_ASSERT(libfab_tm_is_locked(tm->ntm_xprt_private));
+	net = m0_tl_find(m0_nep, net, &tm->ntm_end_points,
+			 libfab_ep_cmp(libfab_ep(net), name, &ep_name_n));
+
+	if (net == NULL) {
+		if (name != NULL)
+			rc = libfab_ep_create(tm, name, epn, epp);
+		else {
+			memset(&ep_str, 0, sizeof(ep_str));
+			M0_ASSERT(epn != NULL);
+			M0_ASSERT((strlen(epn->fen_addr) + strlen(epn->fen_port)
+				  + 8) < LIBFAB_ADDR_STRLEN_MAX);
+			sprintf(ep_str, "libfab:%s:%s", epn->fen_addr,
+				epn->fen_port);
+			rc = libfab_ep_create(tm, ep_str, epn, epp);
+		}
+	} else {
+		ep = libfab_ep(net);
+		*epp = &ep->fep_nep;
+		if (name != NULL && epn != NULL) {
+			wc = strchr(name,'*');
+			if (wc != NULL &&
+			    strcmp(ep->fep_name_p.fen_port, epn->fen_port)
+									 != 0) {
+				strcpy(ep->fep_name_p.fen_addr, epn->fen_addr);
+				strcpy(ep->fep_name_p.fen_port, epn->fen_port);
+				libfab_ep_pton(&ep->fep_name_p,
+					       &ep->fep_name_n);
+				aep = libfab_aep_get(ep);
+				ma = tm->ntm_xprt_private;
+				if (aep->aep_tx_state == FAB_CONNECTED)
+					rc = libfab_txep_init(aep, ma, ep);
+			}
```

Comment:
In order to support lnet addressing format, where we might get a '*' (auto assign) instead of a numeric value for tmid, this is required.
Added a comment

Reply:
Why sock does not need logic like this?